src/utils/references.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2024 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+// eslint-disable-next-line import/prefer-default-export
+export function shouldInactivateRef(identifier, context = {}) {
+  const { includedArchiveIdentifiers = [] } = context;
+  return !!(includedArchiveIdentifiers.length && (
+    !includedArchiveIdentifiers.some(id => (
+      identifier?.startsWith(`doc://${id}`)
```

Comment:
1. Capitalization would be matched exactly as-is here. I don't think we would want to be less strict since we're given the exact identifier.
2. That would only work for documentation pages and not tutorial pages. We could make this more strict, but I think checking for the special scheme and bundle ID should be sufficient (@d-ronnqvist correct me if I'm wrong)